python/lsst/eotest/image_utils.py

```diff
@@ -78,56 +78,210 @@ def dm_hdu(hdu):
+    returns a numpy.poly1d object with the fitted bias as function of pixel row. 
+    Allows the option to explicitly set the fit order to apply to 
+    each row using additional **kwargs.
+
+    Args:
+        im: A masked (lsst.afw.image.imageLib.MaskedImageF) or unmasked 
+            (lsst.afw.image.imageLib.ImageF) afw image.
+        overscan: A bounding box for the serial overscan region.
+        dxmin: The number of columns to skip at the beginning of the serial
+            overscan region.
+        dxmax: The number of columns to skip at the end of the serial overscan region.
+        statistic: The statistic to use to calculate the offset for each row.   
+ 
+    Keyword Arguments:
+        fit_order: The order of the polynomial. The default is: 1.
+
+    Returns:
+        A np.poly1d object containing the coefficients for the polynomial fit.
+    """
+    try:
+        imarr = im.Factory(im, overscan).getArray()
+    except AttributeError: # Dealing with a MaskedImage
+        imarr = im.Factory(im, overscan).getImage().getArray()
+    ny, nx = imarr.shape
+    rows = np.arange(ny)
+    values = np.array([statistic(imarr[j][dxmin:-dxmax]) for j in rows])
+    return np.poly1d(np.polyfit(rows, values, kwargs.get('fit_order', 1))) 
+
+def bias_spline(im, overscan, dxmin=5, dxmax=2, statistic=np.mean, **kwargs):
+    """Compute the offset by fitting a spline to the mean of each row in the 
+    serial overscan region.
+
+    Args:
+        im: A masked (lsst.afw.image.imageLib.MaskedImageF) or unmasked
+            (lsst.afw.image.imageLib.ImageF) afw image.
+        overscan: A bounding box for the serial overscan region.
+        dxmin: The number of columns to skip at the beginning of the serial
+            overscan region.
+        dxmax: The number of columns to skip at the end of the serial overscan region.
+        statistic: The statistic to use to calculate the offset for each row.
+
+    Keyword Arguments:
+        k: The degree of the spline fit. The default is: 3.
+        s: The amount of smoothing to be applied to the fit. The default is: 18000.
+        t: The number of knots. If None, finds the number of knots to use 
+            for a given smoothing factor, s. The default is: None.
+
+    Returns:
+        A tuple (t,c,k) containing the vector of knots, the B-spline coefficients, 
+        and the degree of the spline.
+    """ 
 
-def bias_image(im, overscan, fit_order=1, statistic=np.mean):
-    my_bias = bias_func(im, overscan, fit_order, statistic=statistic)
+    try:
+        imarr = im.Factory(im, overscan).getArray()
+    except AttributeError: # Dealing with a MaskedImage
+        imarr = im.Factory(im, overscan).getImage().getArray() 
+    ny, nx = imarr.shape
+    rows = np.arange(ny)
+    values = np.array([statistic(imarr[j][dxmin:-dxmax]) for j in rows])
+    rms = 7 # Expected read noise per pixel
+    weights = np.ones(ny) * (rms / np.sqrt(nx))
```

Comment:
Since the pixel values in the overscan region have stdev = read noise, I think the error on the mean of any number of pixels is actually just the read noise, so given the `scipy.interpolate.splrep` documentation, I think this should be 
```
weights = np.ones(ny)*rms
```

Reply:
So the error on the mean would be independent of the number of columns we are averaging over?

Reply:
oops, sorry, you're right.  It should be `rms/np.sqrt(nx)`.